Create a new release 1.7.0

JafarAbdi opened the pull request:
@MarqRazz and I are working on cutting a new release to Jazzy/Rolling, this is our first attempt to take over the release process for ROS 2.

@mamoll We have a few questions:

- How do you organize the release notes? I just copied commits' since the last release for now
- How do you want to handle tagging the release? Pushing the git tag
- Should we bump the version number after we tag? To make it so building from source has a different version from the binaries

JafarAbdi commented on doc/markdown/releaseNotes.md:
```diff
@@ -1,5 +1,53 @@
 
+## OMPL 1.7.0 (December 20, 2024)
+
+- Fix cmake installation with version folder 
```
I removed the CI related ones

mamoll commented:
@JafarAbdi / @MarqRazz, thanks for getting the ball rolling for a 1.7.0 release. An OMPL release is separate from a release for ROS2. There are several downstream maintainers (like yourself) who release OMPL for: ROS2, the "official" Debian/Ubuntu package, vcpkg, Fedora, Homebrew, Macports, etc. It seems like enough has changed that perhaps it's a good time to cut a new release. @zkingston @wbthomason @kylc : are there any outstanding issues that need to be fixed first? I am thinking of Python binding issues, but there could be others.

To answer you question re. release process, this is the checklist we put together many years ago that roughly describes this:
https://github.com/ompl/omplapp/blob/main/doc/releaseChecklist.txt

MarqRazz commented:
Would it be possible for us to make some changes to the release process? I think it would be clearer if we were to bump the version number after each release. That way from source or binary it is obvious which version you are working with and know if the feature/patch you need has been released.

mamoll commented:
> Would it be possible for us to make some changes to the release process? I think it would be clearer if we were to bump the version number after each release. That way from source or binary it is obvious which version you are working with and know if the feature/patch you need has been released.

I think we already bump the version number for every release. Is this not clear in the release check list? Or have we not followed the check list in the past? I agree that stealth updates should be avoided.

MarqRazz commented:
Sorry I'm suggesting we bump the source code after each release so that it does not match the last tag. I found it confusing that `1.6.0` was released in January 2023 and source still had the same version number but had breaking changes merged into it. If we were to move bumping the number to after we tag then it would be obvious which version you are working with.

simonschmeisser commented:
There's at least one open MoveIt PR that could have been merged if there were a new version number already, so +1 for early bumping from me

MarqRazz commented:
Is there anything I can do to help move this along?

kylc commented:
As far as I know there isn't any blocker on the Python wheel generation. The wheel CI jobs on the latest main branch commit were successful.

One nice-to-have before a release would be to upgrade to boost-python 1.87 to enable numpy 2.0 support (see https://github.com/boostorg/python/issues/431). Currently, if you use the CI wheels, you have to install `numpy<2.0`.

kylc commented on CMakeLists.txt:
```diff
@@ -1,6 +1,6 @@
 cmake_minimum_required(VERSION 3.12)
-project(ompl VERSION 1.6.0 LANGUAGES CXX)
-set(OMPL_ABI_VERSION 17)
+project(ompl VERSION 1.7.0 LANGUAGES CXX)
```
Also bump the version in [py-bindings/setup.py](https://github.com/ompl/ompl/blob/d3f2cc76de0dfc33d1e6322adb2376b90e1c1241/py-bindings/setup.py#L153)

JafarAbdi commented:
> I'd also like [my own recent PR](https://github.com/ompl/ompl/pull/1210) merged in. That shouldn't affect much else in OMPL (or MoveIt).

Anything I can help with? I fixed an issue with Python wheel generation and added python-binding for the new state spaces [here](https://github.com/ompl/ompl/pull/1215)

MarqRazz commented:
Bumping this to see how it's coming along. What do you think about making a list of blocking PR's here so that we can start to track progress towards releasing?

lepalom commented:
@mamoll If you release a new version I would try to put it on Debian ASAP. I know that most of you use the ROS2 version, but the original Debian package doesn't depend on ROS and can be interesting for some people.

MarqRazz commented:
Any movement here? Do we have a list of what is blocking us from releasing?

mamoll commented:
Hi @MarqRazz, sorry it's taking so long. We are very close. I have a PR (PR #1242) with some cleanup in preparation of release. I am handing over the actual release process to @saahu27, a new maintainer and software engineer in Lydia Kavraki's lab. @saahu27, do you have an estimate of how much more time you need?

saahu27 commented:
Hi @MarqRazz  @mamoll I'll need about 1-2 weeks to complete the release process. I'm currently getting up to speed with all the necessary procedures and working through some parallel tasks. I should have a more specific timeline by the end of next week.

Please let me know if you need any additional information from me in the meantime or if there are any specific priorities I should focus on for this release.

MarqRazz commented:
> Please let me know if you need any additional information from me in the meantime or if there are any specific priorities I should focus on for this release.

[This](https://github.com/ompl/ompl/pull/1185) is the only PR that I can think of. If @Ryanf55 is able to build with the legacy CMAKE variables then I think we should get it in and not worry about MoveIt (we don't need legacy any more). If we get that fixed down the road it should be an easy patch to extend legacy support.

mamoll commented:
release is out, so closing this issue. Please re-open if there are issues with it.
